helm/resource_release.go

```diff
@@ -312,8 +312,13 @@ func resourceDiff(d *schema.ResourceDiff, meta interface{}) error {
 		return nil
 	}
 
-	// Set desired version from the Chart metadata
-	return d.SetNew("version", c.Metadata.Version)
+	// Set desired version from the Chart metadata if available
+	if len(c.Metadata.Version) > 0 {
+		return d.SetNew("version", c.Metadata.Version)
+	} else {
+		return d.SetNewComputed("version")
```

Comment:
Hi @alexsomesan, what do you think about that change to try and make the provider behave more reliably?

This apparently has resolved the flakiness of the newly added acceptance test: https://github.com/terraform-providers/terraform-provider-helm/pull/273#issuecomment-492181707

Note: I did not manage to reproduce the issue locally.